File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_040)_Print_N_bonacci_Numbers.cpp
#include <iostream>
using namespace std;

void bonacciseries(long n, int m)
{
	cout << "N-bonacci Numbers = ";
	int a[m] = {0};
	a[n - 1] = 1;
	a[n] = 1;
	for (int i = n + 1; i < m; i++)
		a[i] = 2 * a[i - 1] - a[i - n - 1];
	for (int i = 0; i < m; i++)
		cout << a[i] << " ";
	cout << endl;
}

int main()
{
	int N = 5, M = 15;
	bonacciseries(N, M);
	return 0;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_040)_Print_N_bonacci_Numbers.dart
void printNbonacciNumbers(int n, int m) {
  var series = List.generate(m, (i) => 0);
  series[n - 1] = 1;
  series[n] = 1;
  for (int i = n + 1; i < m; i++) {
    series[i] = 2 * series[i - 1] - series[i - n - 1];
  }
  print('N-bonacci Numbers = $series.');
}

void main() {
  int n = 5;
  int m = 15;
  printNbonacciNumbers(n, m);
}
